app/js/main.js

```diff
@@ -749,112 +762,157 @@ function notifyError(msg) {
 }
 
 /**
- * Display top menu
+ * Display a custom confirm message
+ *
+ * @param {String|Nunber} [msg=""] The confirm message to display.
  */
-// Create menu
-var menuBar = new gui.Menu({ type: 'menubar' });
-
-// Create sub-menus
-var fileMenuItems = new gui.Menu(),
-    editMenuItems = new gui.Menu(),
-    captureMenuItems = new gui.Menu(),
-    helpMenuItems = new gui.Menu();
-
-//File menu items
-fileMenuItems.append(new gui.MenuItem({
-  label: "New project...",
-    icon: "pngicons/file.png",
-  click: function() {
-  },
-    key: "n",
-    modifiers: "ctrl",
-}));
-fileMenuItems.append(new gui.MenuItem({
-  label: "Open project...",
-    icon: "pngicons/import.png",
-  click: function() {
-  },
-    key: "o",
-    modifiers: "ctrl",
-}));
-fileMenuItems.append(new gui.MenuItem({
-  label: "Main Menu",
-    icon: "pngicons/menu.png",
-  click: function() {
-    openIndex();
-  },
-    key: "m",
-    modifiers: "ctrl",
-}));
-
-//Edit menu items
-editMenuItems.append(new gui.MenuItem({
-  label: "Delete last frame",
-    icon: "pngicons/delete.png",
-  click: function() {
-    undoFrame();
-  },
-  key: "z",
-  modifiers: "ctrl",
-}));
-
-//Capture menu items
-captureMenuItems.append(new gui.MenuItem({
-  label: "Capture frame",
-    icon: "pngicons/capture.png",
-  click: function() {
-    takePicture();
-  },
-  key: "c",
-  modifiers: "ctrl",
-}));
-
-//Help menu items
-helpMenuItems.append(new gui.MenuItem({
-  label: "Give feedback",
-    icon: "pngicons/feedback.png",
-  click: function() {
-      gui.Shell.openExternal('https://github.com/BoatsAreRockable/animator/issues');
-  },
-  key: "/",
-  modifiers: "ctrl",
-}));
-
-// Append sub-menus to main menu
-menuBar.append(
-    new gui.MenuItem({
-        label: 'File',
-        submenu: fileMenuItems
-    })
-);
-menuBar.append(
-    new gui.MenuItem({
-        label: 'Edit',
-        submenu: editMenuItems
-    })
-);
-menuBar.append(
-    new gui.MenuItem({
-        label: 'Capture',
-        submenu: captureMenuItems
-    })
-);
-menuBar.append(
-    new gui.MenuItem({
-        label: 'Help',
-        submenu: helpMenuItems
-    })
-);
-
-// Append main menu to Window
-gui.Window.get().menu = menuBar;
-
-// Create Mac menu
-if (process.platform === "darwin") {
-    menuBar.createMacBuiltin('Boats Animator');
+function confirmSet(action, args, msg) {
+    "use strict";
+    confirmText.innerHTML = msg;
+    confirmContainer.classList.remove("hidden");
+        
+    // Listen if "OK" is pressed
+    confirmOK.addEventListener("click", function() {
+        window[action](args);
```

Comment:
Functions in JS are first-class citizens. You can just pass a reference to a function and call it. Change the code to:

```js
function confirmSet(func, args, msg) {
    // Same code
    confirmOK.addEventListener("click", function() {
        if (args === undefined) {
            func();
        } else {
            func(args);
        }
    });
}
```

Obviously properly indent as needed. It may be possible to do this with a ternary operator but I'm multitasking and can't review this in-depth right now.